packages/camera/camera/ios/Classes/FLTSavePhotoDelegate.m

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTSavePhotoDelegate.h"
+
+@implementation FLTSavePhotoDelegate
+
+- (instancetype)initWithPath:(NSString *)path
+                      result:(FLTThreadSafeFlutterResult *)result
+                     ioQueue:(dispatch_queue_t)ioQueue {
+  self = [super init];
+  NSAssert(self, @"super init cannot be nil");
+  _path = path;
+  _selfReference = self;
+  _result = result;
+  _ioQueue = ioQueue;
+  return self;
+}
+
+- (void)handlePhotoCaptureResultWithError:(NSError *)error
+                        photoDataProvider:(NSData * (^)(void))photoDataProvider {
+  self.selfReference = nil;
+  if (error) {
+    [self.result sendError:error];
+    return;
+  }
+  dispatch_async(self.ioQueue, ^{
+    NSData *data = photoDataProvider();
+    bool success = [data writeToFile:self.path atomically:YES];
+
+    if (!success) {
+      [self.result sendErrorWithCode:@"IOError" message:@"Unable to write file" details:nil];
+      return;
+    }
+    [self.result sendSuccessWithData:self.path];
```

Comment:
Can you make all references to `bool` -> `BOOL`?

Probably makes to send back the error details, something like
```dart
    NSError *error;
    if ([data writeToFile:self.path options:NSDataWritingAtomic error:&error]) {
      [self.result sendSuccessWithData:self.path];
    } else {
      [self.result sendErrorWithCode:@"IOError" message:@"Unable to write file" details:error.localizedDescription];
    }
```

Reply:
Sharp eyes! I didn't notice the `bool`.